src/mgmt/srv_drpc.c

```diff
@@ -565,6 +565,67 @@ ds_mgmt_drpc_pool_destroy(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
 	mgmt__pool_destroy_req__free_unpacked(req, NULL);
 }
 
+void
+ds_mgmt_drpc_pool_reintegrate(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
+{
+	Mgmt__PoolReintegrateReq	*req = NULL;
+	Mgmt__PoolReintegrateResp	resp = MGMT__POOL_REINTEGRATE_RESP__INIT;
+	uuid_t				uuid;
+	struct pool_target_id_list	reint_list;
+	uint32_t			reint_rank;
+	uint8_t				*body;
+	size_t				len;
+	int				rc, i;
+
+	/* Unpack the inner request from the drpc call body */
+	req = mgmt__pool_reintegrate_req__unpack(
+		NULL, drpc_req->body.len, drpc_req->body.data);
+
+	if (req == NULL) {
+		drpc_resp->status = DRPC__STATUS__FAILED_UNMARSHAL_PAYLOAD;
+		D_ERROR("Failed to unpack req (set up pool target)\n");
```

Comment:
Is "set up pool target" an accurate description of this failure? "Reintegrate pool target" may be a better description.